File: exercise/day01/src/main/java/food/Food.java
package food;

import java.time.LocalDate;
import java.util.UUID;
import java.util.function.Supplier;

public record Food(LocalDate expirationDate,
                   Boolean approvedForConsumption,
                   UUID inspectorId) {
    public boolean isEdible(Supplier<LocalDate> now) {

        if (inspectorId == null) return false;

        boolean isNotExpired = expirationDate.isAfter(now.get());

        return isNotExpired && approvedForConsumption;
    }
}

File: exercise/kotlin/day01/src/main/kotlin/food/Food.kt
package food

import java.time.LocalDate
import java.util.*

data class Food(val expirationDate: LocalDate, val approvedForConsumption: Boolean, val inspectorId: UUID?) {
    fun isEdible(now: () -> LocalDate): Boolean {
        return if (expirationDate.isAfter(now()) &&
            approvedForConsumption &&
            inspectorId != null
        ) {
            true
        } else {
            false
        }
    }
}